gr-trellis/include/gnuradio/trellis/metrics.h

```diff
@@ -36,28 +36,33 @@ namespace gr {
      * \brief Evaluate metrics for use by the Viterbi algorithm.
      * \ingroup trellis_coding_blk
      */
-    class TRELLIS_API @NAME@ : virtual public block
+template<class T>
+    class TRELLIS_API metrics : virtual public block
     {
     public:
-      // gr::trellis::@BASE_NAME@::sptr
-      typedef boost::shared_ptr<@BASE_NAME@> sptr;
+      // gr::trellis::metrics::sptr
+      typedef boost::shared_ptr< metrics<T> > sptr;
 
-      static sptr make(int O, int D, const std::vector<@I_TYPE@> &TABLE,
+      static sptr make(int O, int D, const std::vector<T> &TABLE,
 		       digital::trellis_metric_type_t TYPE);
 
       virtual int O() const = 0;
       virtual int D() const = 0;
       virtual digital::trellis_metric_type_t TYPE() const = 0;
-      virtual std::vector<@I_TYPE@> TABLE() const = 0;
+      virtual std::vector<T> TABLE() const = 0;
 
       virtual void set_O(int O) = 0;
       virtual void set_D(int D) = 0;
       virtual void set_TYPE(digital::trellis_metric_type_t type) = 0;
-      virtual void set_TABLE(const std::vector<@I_TYPE@> &table) = 0;
+      virtual void set_TABLE(const std::vector<T> &table) = 0;
     };
 
 
+typedef metrics<std::int16_t> metrics_s;
+typedef metrics<std::int32_t> metrics_i;
+typedef metrics<float> metrics_f;
+typedef metrics<gr_complex> metrics_c;
```

Comment:
`calc_metric` exists for `char`. Please add an instantiation for char, if possible